tools/xdp/xtask/src/ci.rs

```diff
@@ -39,3 +45,44 @@ fn dump() -> Result<String, anyhow::Error> {
     let output = String::from_utf8(output)?;
     Ok(output)
 }
+
+fn test() -> Result<(), anyhow::Error> {
+    let status = Command::new("cargo")
+        .arg("test")
+        .status()
+        .expect("failed to run tests");
+    assert!(status.success());
+    Ok(())
+}
+
+fn cap_net_raw_tests() -> Result<(), anyhow::Error> {
+    let mut at_least_one = false;
+
+    for file in std::fs::read_dir("target/debug/deps")?.flatten() {
+        let path = file.path();
+
+        if path.extension().is_some()
+            || !path
+                .file_name()
+                .and_then(|name| name.to_str())
+                .map_or(false, |name| name.starts_with("s2n_quic_xdp"))
+        {
+            continue;
+        }
+
+        let status = Command::new("sudo")
+            .arg(Path::new("../").join(path))
+            .arg("--nocapture")
+            .current_dir("s2n-quic-xdp")
+            .env("CAP_NET_RAW_ENABLED", "1")
+            .status()
+            .expect("failed to run test case");
+        assert!(status.success());
```

Comment:
So our CI can support sudo and AF_XDP? Also is this command missing a `cargo test`? Not sure what its running.

Reply:
Yes you can see the CI task is passing with this. It doesn't need a `cargo test` because it's invoking the test binary directly. Running `sudo cargo test` makes a mess of permissions and ends up fetching a new registry and recompiling the whole thing.

Reply:
> WARNING: CAP_NET_RAW capabilities missing; skipping `syscall_test`

I dont think its being tested in our CI atm.

Reply:
It is. First it tests it without. Then it runs it with.

Reply:
https://github.com/aws/s2n-quic/actions/runs/4673058844/jobs/8275868977?pr=1704#step:7:234

Reply:
https://github.com/aws/s2n-quic/blob/7b998791b2f6445648b5bbcec86b80b634ed056c/tools/xdp/xtask/src/ci.rs#L31-L35